milli/src/search/mod.rs

```diff
@@ -140,7 +144,10 @@ impl<'a> Search<'a> {
             None => self.perform_sort(NoopDistinct, matching_words, criteria),
             Some(name) => {
                 let field_ids_map = self.index.fields_ids_map(self.rtxn)?;
-                let id = field_ids_map.id(name).expect("distinct not present in field map");
+                let id = field_ids_map.id(name).ok_or_else(|| FieldIdMapMissingEntry::FieldName {
+                    field_name: name.to_string(),
+                    process: "fetching distint attribute",
```

Comment:
```suggestion
                    process: "distinct attribute",
```
would be displayed as `"unknown [..] coming from the {distinct attribute} process"`